### PR TITLE
prevent "stack init" from clobbering stack without "stack init -f"

### PR DESCRIPTION
**Pull request by displague**
It is possible to run `kubectl crossplane stack init` from the wrong path or as a repeated command and lose all of the existing stack information.

This PR adds a `init -f` argument that must be supplied to allow an existing stack directory to be re-init'd.

This PR also includes a change to `make install` which permits `make install` to be re-run (useful during local development iterations).

### CODE REVIEW

File: bin/kubectl-crossplane-stack-init [hunk 1]
```diff
@@ -3,19 +3,28 @@
 set -e
 
 function usage {
-  echo "Usage: kubectl crossplane stack init STACK_NAME [DIRECTORY]" >&2
+  echo "Usage: kubectl crossplane stack init [-f] STACK_NAME [DIRECTORY]" >&2
```
Comment (suskin) [inline bin/kubectl-crossplane-stack-init]: Could you add support for `--force` too?

File: bin/kubectl-crossplane-stack-init [hunk 2]
```diff
@@ -3,19 +3,28 @@
...
   echo "" >&2
   echo "STACK_NAME is used as the image name for the stack, and any '/' characters" >&2
   echo "are converted to '-' characters when populating kubernetes resource field names." >&2
   echo "" >&2
   echo "DIRECTORY defaults to the current directory." >&2
+  echo "" >&2
+  echo "-f force init when a previous 'stack init' is detected"
```
Comment (suskin) [inline bin/kubectl-crossplane-stack-init]: * This should be echoed to `stderr`
* Could you reword this to sound more like a sentence, like the other usage statements? For example, `-f/--force forces initialization, even when a previous . . . `

File: bin/kubectl-crossplane-stack-init [hunk 3]
```diff
@@ -310,7 +319,13 @@ EOF
   fi
 }
 
-cd ${INIT_DIR}
+cd -- "${INIT_DIR}"
+
+if [ -f "${INIT_DIR}/stack.env" -a -z "${FORCE_INIT}" ]; then
```
Comment (suskin) [inline bin/kubectl-crossplane-stack-init]: Would you mind moving this into a function? I have been trying to keep the main section of the script clean, using only functions. A potential name could be `check_for_existing_stack` or something similar

File: Makefile [hunk 4]
```diff
@@ -5,7 +5,7 @@ default:
 .PHONY: default
 
 install:
-	ln -s $(abspath bin/kubectl-crossplane-stack-*) $(INSTALL_DIR)/
+	ln -sf $(abspath bin/kubectl-crossplane-stack-*) $(INSTALL_DIR)/
```
Comment (suskin) [inline Makefile]: Since we are adding a `-f` for the main init, I would probably prefer `-i` here instead of `-f`, so that it asks for confirmation to clobber

**Review by suskin — CHANGES_REQUESTED** [review]
Awesome!! Thanks @displague !

I am requesting changes since I had some amendments I was asking for.

**Comment by suskin** [comment]
I forgot to include this in my previous batch of comments, but could you also update the commit messages to follow the standard [crossplane convention](https://github.com/crossplaneio/crossplane/blob/master/CONTRIBUTING.md#commits)?

File: bin/kubectl-crossplane-stack-init (hunk 1 above)
Comment (suskin) [inline bin/kubectl-crossplane-stack-init]: Could you add `--force` to the usage prompt?

File: bin/kubectl-crossplane-stack-init [hunk 5]
```diff
@@ -110,6 +119,13 @@ EOF
   echo 'Created config/stack/overrides/install.yaml' >&2
 }
 
+function check_for_existing_stack {
+  if [ -f "${INIT_DIR}/stack.env" -a -z "${FORCE_INIT}" ]; then
+    echo "${INIT_DIR}/stack.env already exists.  Use -f to force init." >&2
```
Comment (suskin) [inline bin/kubectl-crossplane-stack-init]: Could you add a prefix like `"ERROR: ${INIT_DIR} . . . ` so it's more obvious when scanning that the init failed?

File: bin/kubectl-crossplane-stack-init [hunk 6]
```diff
@@ -3,19 +3,28 @@
...
 function usage {
-  echo "Usage: kubectl crossplane stack init STACK_NAME [DIRECTORY]" >&2
+  echo "Usage: kubectl crossplane stack init [-f] STACK_NAME [DIRECTORY]" >&2
   echo "" >&2
   echo "STACK_NAME is used as the image name for the stack, and any '/' characters" >&2
   echo "are converted to '-' characters when populating kubernetes resource field names." >&2
   echo "" >&2
   echo "DIRECTORY defaults to the current directory." >&2
+  echo "" >&2
+  echo "-f, --force force init when a previous 'stack init' is detected" >&2
```
Comment (suskin) [inline bin/kubectl-crossplane-stack-init]: Could you put some sort of delimiter between the argument and the explanation so this line is easier to scan? For example: `-f, --force : force init . . .`

**Review by suskin — CHANGES_REQUESTED** [review]
Looking pretty good other than a couple minor requests! I'm also liking the nice detailed commit message 🏆

File: bin/kubectl-crossplane-stack-init (hunk 6 above)
Comment (displague) [reply bin/kubectl-crossplane-stack-init]: I was trying to follow the format of a well-known command. In this case `ls`.

I think your last few requests highlight that I've been targeting the wrong tool, and for these requests to seem less arbitrary we should standardize on the most relevant tool: `kubectl`.

For example:

```
kubectl delete --help
```

```
$ kubectl command --help
Command does thing by thing thinger.

 ... length usage notes

 IMPORTANT: ... warnings and notes

Examples:
  # description of command
  kubectl ...arguments
  
  # .. more examples

Options:
  -O, --option=default: description
    ...

Usage:
  kubectl command ([-O ARGUMENT] | [-x X] | TYPE [(NAME | -l label | --all)]) [options]

Use "kubectl options" for a list of global command-line options (applies to all commands).
```

File: bin/kubectl-crossplane-stack-init (hunk 5 above)
Comment (displague) [reply bin/kubectl-crossplane-stack-init]: How about following this example ([see my other comment for context](https://github.com/crossplaneio/crossplane-cli/pull/6#discussion_r321741824)).
```
$ kubectl foo  --help
Error: unknown command "foo" for "kubectl"
```

Comment (suskin) [reply bin/kubectl-crossplane-stack-init]: I like these ideas! How about if we say we want this to be the style we're going for, and as we edit lines, we update them to match the style?

**Review by suskin — APPROVED** [review]
Awesomeeee :shipit: :ship:
